privacyidea/api/smsgateway.py

```diff
@@ -124,21 +129,23 @@ def delete_gateway(identifier=None):
     return send_result(res)
 
 
-@smsgateway_blueprint.route('/option/<gwid>/<option>', methods=['DELETE'])
+@smsgateway_blueprint.route('/<type>/<gwid>/<key>', methods=['DELETE'])
 @log_with(log)
 @prepolicy(check_base_action, request, ACTION.SMSGATEWAYWRITE)
-def delete_gateway_option(gwid=None, option=None):
+def delete_gateway_option(gwid=None, key=None, type=None):
     """
     this function deletes an option of a gateway definition
 
     :param gwid: The id of the sms gateway definition
     :return: json with success or fail
```

Comment:
Please add in the docstring, that <type> is either "option" or "header", since the documentation for the endpoint

 DEL /option/

will disappear.
So you might also check again, if two bluprints (one for /option and one for /header) work.